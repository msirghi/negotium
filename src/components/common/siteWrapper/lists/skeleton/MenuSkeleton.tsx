import { Skeleton } from '@mui/material';
import { Box } from '@mui/system';
import { memo } from 'react';

const skeletonHeight = 30;

export const MenuSkeleton = () => {
  return (
    <Box sx={{ padding: 1.5 }}>
      {[1, 2, 3, 4].map((key) => (
        <Skeleton key={key} variant="text" height={skeletonHeight} />
      ))}
    </Box>
  );
};

export default memo(MenuSkeleton);
