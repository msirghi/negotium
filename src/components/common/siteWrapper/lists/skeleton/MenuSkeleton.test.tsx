import renderer from 'react-test-renderer';
import { MenuSkeleton } from './MenuSkeleton';
import { mount } from 'enzyme';
import { Skeleton } from '@mui/material';

describe('MenuSkeleton', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<MenuSkeleton />);
    expect(tree).toMatchSnapshot();
  });

  it('should render 4 skeletons', () => {
    const wrapper = mount(<MenuSkeleton />);
    expect(wrapper.find(Skeleton)).toHaveLength(4);
  });
});
