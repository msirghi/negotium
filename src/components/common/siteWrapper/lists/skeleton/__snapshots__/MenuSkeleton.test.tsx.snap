// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSkeleton should match the snapshot 1`] = `
<div
  className="MuiBox-root css-38zrbw"
>
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    style={
      Object {
        "height": 30,
        "width": undefined,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    style={
      Object {
        "height": 30,
        "width": undefined,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    style={
      Object {
        "height": 30,
        "width": undefined,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    style={
      Object {
        "height": 30,
        "width": undefined,
      }
    }
  />
</div>
`;
